crates/api_common/src/utils.rs

```diff
@@ -158,6 +152,26 @@ pub async fn mark_post_as_read(
   Ok(())
 }
 
+/// Updates the read comment count for a post. Usually done when reading or creating a new comment.
+#[tracing::instrument(skip_all)]
+pub async fn update_read_comments(
```

Comment:
Made a helper function for this, since its used now both when you read the post, and when you create a new comment.